src/main/java/org/openrewrite/java/dependencies/github/ParseAdvisories.java

```diff
@@ -40,50 +39,93 @@
 
 public class ParseAdvisories {
     public static void main(String[] args) throws IOException {
-        ObjectMapper mapper = new ObjectMapper()
+        if (args.length != 2) {
+            System.err.println("Usage: ParseAdvisories <advisories-repo> <advisories-csv>");
+            System.exit(1);
+        }
+        File advisoriesRepo = new File(args[0]);
+        if (!advisoriesRepo.isDirectory() || !advisoriesRepo.canRead()) {
+            System.err.println("Advisories repo " + advisoriesRepo + " not readable");
+            System.exit(1);
+        }
+        File advisoriesCsv = new File(args[1]);
+        if (!advisoriesCsv.createNewFile() && !advisoriesCsv.canWrite()) {
+            System.err.println("Advisories CSV " + advisoriesCsv + " not writable");
+            System.exit(1);
+        }
+
+        parseAdvisories(advisoriesRepo, advisoriesCsv);
+    }
+
+    static void parseAdvisories(File advisoriesRepoInput, File advisoriesCsvOutput) throws IOException {
+        try (FileOutputStream fos = new FileOutputStream(advisoriesCsvOutput)) {
+            Files.walkFileTree(advisoriesRepoInput.toPath(), emptySet(), 16, new PathSimpleFileVisitor(fos));
+        }
+    }
+}
+
+class PathSimpleFileVisitor extends SimpleFileVisitor<Path> {
```

Comment:
why not make this a private static inner class? Also the naming is not very revealing... Maybe something related to Advisories?

Reply:
Good suggestion, thanks. Adopted in https://github.com/openrewrite/rewrite-java-dependencies/pull/19/commits/a9850ed8bbaacdfc379e5e794a568d5ea36d375c